### ISSUE TITLE
In-Game: Error when trying to call function using its index stored in an instance array and accessed via "self"

### ISSUE THREAD

**Issue by biyectivo** [YoYoGames/GameMaker-Bugs#2065]
### Description

(Sorry for the somewhat involved title)

This is a fairly specific bug that occurs when one has a function index stored in an array (for example, this happens in my UI library, *gooey*, where the user specifies callback functions that get called whenever a certain interaction event happens, such as a left mouse released event). When said array is then referenced using `self` and then called in the same line, it throws an error.

<img width="295" alt="image" src="https://github.com/YoYoGames/GameMaker-Bugs/assets/28306587/3ba8c53c-8f98-4a13-8c75-f6922a326bb0">

<img width="978" alt="image" src="https://github.com/YoYoGames/GameMaker-Bugs/assets/28306587/e21b1930-63a2-4969-b924-1f8ab1fc8837">

The built-in function can have 0 or more arguments (with 1 or more arguments, one would obviously need to specify them in the function calls below):

```
self.func = window_center;
```

Code:

```
self.arr = [];
array_push(self.arr, self.func);
arr[0]();						// works
self.arr[0]();						// Error: Unable to get object for index XXXXX
```

It's noteworthy that assigning the (index) value contained in the array to a local variable and then calling the corresponding function, works:

```
var _arr = self.arr[0];
_arr();							// works
```

Also, this behavior does not happen when the index is stored inside an instance or local variable instead of an array:

```
func()							// works
self.func();						// works
```

Lastly, none of this happens either when using a custom function:

```
function test() {
	show_debug_message("TEST at {0}", current_time);
}

self.func = test;

self.arr = [];
array_push(self.arr, self.func);
arr[0]();						// works
self.arr[0]();						// works as well!
```

The code shown is included in the YYZ attached below for easy replicability

[Function calling bug with self.zip](https://github.com/YoYoGames/GameMaker-HTML5/files/13079404/Function.calling.bug.with.self.zip)

[Function calling bug with self - Contact Us package.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13079462/Function.calling.bug.with.self.-.Contact.Us.package.zip)

[Contact Us Package 2](https://api.gamemaker.io/api/github/downloads/f292f9be-8b55-4695-8086-836d40f17862)

### Steps To Reproduce

1. Start GameMaker
2. Load the sample project
3. Hit F5 / run

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2023.8 (Monthly), 2022.0.1 (LTS), Beta (Please put the exact version number in your Description)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by biyectivo** [comment]
BTW apologies if this is not the right way to reopen closed issues, but the issue persists even in 

IDE 2023.1100.0.425
Runtime 2023.1100.0.439

Please download and run the included project to test it
[Function calling bug with self (2).zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13212199/Function.calling.bug.with.self.2.zip)

**Comment by iampremo** [comment]
> BTW apologies if this is not the right way to reopen closed issues

Since you opened the original issue you should have a reopen button in the comment box at the bottom of the page

![Image](https://github.com/YoYoGames/GameMaker-Bugs/assets/16644676/536fc478-fc42-42a9-a487-605afc6f7cec)

**Comment by biyectivo** [comment]
I think it's for YYG only as I both originally and now searched for the button and it's not there, that's why I resorted to quoting the previous issue in a new issue. 

Thanks for assigning it :)

**Comment by rwkay** [comment]
I have checked with `2023.1100.0.439` and with the current internal Beta version and I cannot reproduce this bug at all... please check that you downloaded the `2023.1100.0.439` runtime properly

**Comment by tinkerer-red** [comment]
Can confirm this is crashing
```

___________________________________________
############################################################################################
ERROR in
action number 1
of Create Event
for object obj_Test:

Unable to get object for index 367
 at gml_Object_obj_Test_Create_0 (line 23) - self.arr[0]();                                          // Error: Unable to get object for index XXXXX
############################################################################################
gml_Object_obj_Test_Create_0 (line 23)
```

**Comment by JujuAdams** [comment]
Beta IDE v2023.1100.0.425, Beta Runtime v2023.1100.0.439, Test | GMS2 VM | Default | Default

```
___________________________________________
############################################################################################
ERROR in
action number 1
of Create Event
for object obj_Test:

Unable to get object for index 367
 at gml_Object_obj_Test_Create_0 (line 23) - self.arr[0]();                                          // Error: Unable to get object for index XXXXX
############################################################################################
gml_Object_obj_Test_Create_0 (line 23)
```

**Comment by rwkay** [comment]
@YYDan can you please get the QA team to check this as I cannot get it replicated at all on my machine, though others definitely can

**Comment by biyectivo** [comment]
I tested on several additional platforms for further reference:

- Windows VM - crash
- Windows YYC - no crash
- HTML5 - no crash
- GX.Games - crash
- Android - crash
- Linux (Raspberry Pi / Raspbian) VM - crash

**Comment by Emc1923** [comment]
Able to repro with Windows GMS2 VM:

`___________________________________________
############################################################################################
ERROR in
action number 1
of Create Event
for object obj_Test:

Unable to get object for index 1150
 at gml_Object_obj_Test_Create_0 (line 19) - self.arr[0]();                                          // Error: Unable to get object for index XXXXX
############################################################################################
gml_Object_obj_Test_Create_0 (line 19)
`

**Comment by rwkay** [comment]
Fixed in 2023.11

**Comment by sihammill** [comment]
Fix verified in IDE v2023.1100.0.436  Runtime v2023.1100.0.449

**Comment by biyectivo** [comment]
Can confirm! Thx
